src/solver/variable/state.h

```diff
@@ -37,23 +37,40 @@
 
 namespace Antares::Solver::Variable
 {
+class ThermalState
```

Comment:
I can't prevent myself from thinking that the computation we do on thermal clusters in **State** class after the optimization has run is an optimization post-treatment. 
Maybe these vectors should join the **problemeHebdo->ResultatsHoraires** and be computed as a post-treatment. 
If this is suitable and doable, the **State** class should shrink dramatically.

Reply:
Yes, the problem is that it's a **yearly** post-treatment, which is why those vectors are of size 8760. Therefore it wouldn't make sense to place it inside problemeHebdo->ResultatsHoraires, since each variable defined there only has size 168 (see e.g src/solver/simulation/economy.cpp:98)